libpf/trace.go

```diff
@@ -10,15 +10,27 @@ package libpf
 // stack frame via the file ID and line number at the offset i in the trace. The
 // information for the most recently called function is at offset 0.
 type Trace struct {
-	Files      []FileID
-	Linenos    []AddressOrLineno
-	FrameTypes []FrameType
-	Hash       TraceHash
+	Files              []FileID
+	Linenos            []AddressOrLineno
+	FrameTypes         []FrameType
+	MappingStart       []Address
+	MappingEnd         []Address
+	MappingFileOffsets []uint64
```

Comment:
This is becoming increasingly ugly: we should switch this from struct-of-array to array-of-struct soon. The reasons for the columnar representation here are long gone.